### PR TITLE
LPS-129709 Add option to invite other users to collaborate on a publication

### PR DESCRIPTION
**Pull request by stsquared99**
@Preston-Crary Can you actually double check this after all?

### CODE REVIEW

**Comment by Preston-Crary** [comment]
This doesn't upgrade correctly because of the permissions we add when we enable publications. checkCompany() is also called after our upgrade so we should handle both missing and existing publication role.

File: modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/upgrade/v2_5_1/PublicationsUserRoleUpgradeProcess.java [hunk 1]
```diff
@@ -1,99 +0,0 @@
...
-	@Override
-	protected void doUpgrade() throws Exception {
-		try (PreparedStatement preparedStatement = connection.prepareStatement(
-				StringBundler.concat(
-					"select CTPreferences.companyId from CTPreferences left ",
-					"join Role_ on Role_.companyId = CTPreferences.companyId ",
-					"and Role_.name = '",
-					CTConstants.ROLE_NAME_PUBLICATIONS_USER,
-					"' where CTPreferences.userId = 0 and Role_.roleId is ",
-					"null"));
```
Comment (Preston-Crary) [inline modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/upgrade/v2_5_1/PublicationsUserRoleUpgradeProcess.java:59]: Keep this upgrade but remove the `and Role_.roleId is null` part of the query.

File: modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/upgrade/v2_5_1/PublicationsUserRoleUpgradeProcess.java [hunk 2]
```diff
@@ -1,99 +0,0 @@
...
-			ResultSet resultSet = preparedStatement.executeQuery()) {
-
-			while (resultSet.next()) {
-				long companyId = resultSet.getLong(1);
-
-				long defaultUserId = _userLocalService.getDefaultUserId(
-					companyId);
-
-				Role role = _roleLocalService.addRole(
```
Comment (Preston-Crary) [inline modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/upgrade/v2_5_1/PublicationsUserRoleUpgradeProcess.java:68]: Fetch the role here and add if it's missing. If it exists add the `CTPortletKeys.PUBLICATIONS` permissions.

**Comment by Preston-Crary** [comment]
Please check my comments and resend. Thanks!

**Comment by stsquared99** [comment]
Resent: https://github.com/Preston-Crary/liferay-portal/pull/1392

**Pull request closed without merging**